relentless/optimize/method.py

```diff
@@ -171,30 +183,139 @@ def optimize(self, objective):
             return None
 
         iter_num = 0
-        converged = False
-        while not converged and iter_num < self.max_iter:
-            res = objective.compute()
-            for x in dvars:
-                x.value -= self.step_size*res.gradient(x)
-            converged = self.has_converged(res)
+        cur_res = objective.compute()
+        while not self.has_converged(cur_res) and iter_num < self.max_iter:
+
+            #steepest descent update
+            for x in dvars: #is there any way to make this into a single line?
+                x.value = cur_res.design_variables[x] - self.step_size*cur_res.gradient[x]
```

Comment:
The design variables are not a KeyedArray, so this cannot be done in one line.

Remove the extra whitespace above though. :)